src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -817,13 +848,17 @@ private void printToPathGraphics(    final PeekGraphics graphics, // Always an a
         // This is called from the native side.
         Runnable r = new Runnable() { public void run() {
             try {
-                SurfaceData sd = CPrinterSurfaceData.createData(page, context); // Just stores page into an ivar
+                double scaleX = getXRes()/ USER_SPACE_DPI;
+                double scaleY = getYRes()/ USER_SPACE_DPI;
+                PageFormat scaledPage = scalePageFormat(page, scaleX, scaleY);
+                SurfaceData sd = CPrinterSurfaceData.createData(scaledPage, context); // Just stores page into an ivar
```

Comment:
This bleeds through in an application visible way.
Take your test case and add this code to the print(..) method and you'll see what I mean.

        GraphicsConfiguration gc = g2.getDeviceConfiguration();
        Rectangle bds = gc.getBounds();
        System.out.println("bds="+bds);

For me it then prints
bds=java.awt.Rectangle[x=0,y=0,width=2550,height=3300]